rescompy/rescompy.py

```diff
@@ -1023,29 +1025,28 @@ def mapper(inputs, outputs):
         # Attempt to use the compiled version.
         # This will fail if train_result.feature_function is not compiled with
         # a numba.jit decorator.
-        use_jit = numba.extending.is_jitted(feature_function) and \
-            numba.extending.is_jitted(mapper)
-        if use_jit:
+        try:
+            # Propagate the reservoir autonomously.
             states, outputs = _get_states_autonomous_jit(inputs, outputs,
                               states, feature_function,
                               mapper, self.A.data, self.A.indices,
                               self.A.indptr, self.A.shape, self.B, self.C,
                               weights, self.leaking_rate)
-        else:
-            msg = "Using non-compiled autonomous state " \
-                      "propagation function. "\
-                      "Use numba.jit() for feature_function "\
-                      " and mapper to improve performance."
+            states = states
+            outputs = outputs
+
+        except:            
```

Comment:
This needs to be more specific (or use the if `is_jitted()` function) because this catches all error that are not necessarily numba related.

Reply:
#6 addresses this issue.
The except / try block is replaced with two blocks, one that attempts to compile a provided function (if it isn't already) and another that attempts to use the function with the compiled _get_states_autonomous function.
Expected behavior was tested with the compilable, compiled, and non-compilable feature funtions, defined as:
def f_compilable(r, u):
    return r
f_compiled = numba.jit(nopython=True, fastmath=True)(f_compilable)
def f_uncompilable(r, u):
    return r[None][0]